website/src/controllers/api-v3/dataexport.js

```diff
@@ -0,0 +1,225 @@
+import { authWithSession } from '../../middlewares/api-v3/auth';
+import cron from '../../middlewares/api-v3/cron';
+import { model as User } from '../../models/user';
+import * as Tasks from '../../models/task';
+import {
+  NotFound,
+} from '../../libs/api-v3/errors';
+import _ from 'lodash';
+import csvStringify from '../../libs/api-v3/csvStringify';
+import moment from 'moment';
+import js2xml from 'js2xmlparser';
+import Pageres from 'pageres';
+import AWS from 'aws-sdk';
+import nconf from 'nconf';
+import got from 'got';
+import Q from 'q';
+
+let S3 = new AWS.S3({
+  accessKeyId: nconf.get('S3:accessKeyId'),
+  secretAccessKey: nconf.get('S3:secretAccessKey'),
+});
+const S3_BUCKET = nconf.get('S3:bucket');
+
+const BASE_URL = nconf.get('BASE_URL');
+
+let api = {};
+
+/**
+ * @api {get} /export/history.csv Export user tasks history in CSV format. History is only available for habits and dailys so todos and rewards won't be included
+ * @apiVersion 3.0.0
+ * @apiName ExportUserHistory
+ * @apiGroup DataExport
+ *
+ * @apiSuccess {string} A cvs file
+ */
+api.exportUserHistory = {
+  method: 'GET',
+  url: '/export/history.csv',
+  middlewares: [authWithSession, cron],
+  async handler (req, res) {
+    let user = res.locals.user;
+
+    let tasks = await Tasks.Task.find({
+      userId: user._id,
+      type: {$in: ['habit', 'daily']},
+    }).exec();
+
+    let output = [
+      ['Task Name', 'Task ID', 'Task Type', 'Date', 'Value'],
+    ];
+
+    tasks.forEach(task => {
+      task.history.forEach(history => {
+        output.push([
+          task.text,
+          task._id,
+          task.type,
+          moment(history.date).format('YYYY-MM-DD HH:mm:ss'),
+          history.value,
+        ]);
+      });
+    });
+
+    res.set({
+      'Content-Type': 'text/csv',
+      'Content-disposition': `attachment; filename=habitica-tasks-history.csv`,
+    });
+
+    let csvRes = await csvStringify(output);
+    res.status(200).send(csvRes);
+  },
+};
+
+// Convert user to json and attach tasks divided by type
+// at user.tasks[`${taskType}s`] (user.tasks.{dailys/habits/...})
+async function _getUserDataForExport (user) {
+  let userData = user.toJSON();
+  userData.tasks = {};
```

Comment:
I don't know, I thought attaching everything to user.tasks was a good idea but we can change it